src/decorators/docs/course.decorator.ts

```diff
@@ -1,281 +1,61 @@
-    }),
-    ApiResponse({
-      status: 200,
-      description: 'keyword로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  searchMajorCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: 'keyword로 전공 강의 검색',
-      description: 'keyword를 입력하여 전공 강의에서 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'major',
-      required: true,
-      type: 'string',
-    }),
-    ApiResponse({
-      status: 200,
-      description: 'keyword로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-    ApiKukeyExceptionResponse(['MAJOR_REQUIRED']),
-  ],
-  searchGeneralCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: 'keyword로 교양 강의 검색',
-      description: 'keyword를 입력하여 교양 강의에서 검색합니다.',
-    }),
-    ApiResponse({
-      status: 200,
-      description: 'keyword로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  searchAcademicFoundationCourses: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: 'keyword로 학문의 기초 강의 검색',
-      description:
-        'keyword를 입력하여 단과대 별 학문의 기초 강의에서 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'college',
-      required: true,
-      type: 'string',
-    }),
-    ApiResponse({
-      status: 200,
-      description: 'keyword로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-    ApiKukeyExceptionResponse(['COLLEGE_REQUIRED']),
-  ],
-  searchCourseCode: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '학수번호로 강의 검색',
-      description: '학수번호를 입력하여 강의를 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'courseCode',
-      required: true,
-      type: 'string',
+      summary: '강의 검색',
+      description: '하나의 엔드포인트로 모든 강의검색 로직을 통합했습니다.',
     }),
     ApiQuery({
       name: 'cursorId',
       required: false,
       type: 'number',
     }),
-    ApiResponse({
-      status: 200,
-      description: '학수번호로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-  ],
-  searchMajorCourseName: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '전공 과목명 강의 검색',
-      description: '전공 과목명을 입력하여 강의를 검색합니다.',
-    }),
     ApiQuery({
-      name: 'major',
+      name: 'year',
       required: true,
       type: 'string',
     }),
     ApiQuery({
-      name: 'courseName',
+      name: 'semester',
       required: true,
       type: 'string',
     }),
     ApiQuery({
-      name: 'cursorId',
-      required: false,
-      type: 'number',
-    }),
-    ApiResponse({
-      status: 200,
-      description: '전공 과목명으로 강의 검색 성공 시',
-      type: PaginatedCoursesDto,
-    }),
-    ApiKukeyExceptionResponse(['MAJOR_REQUIRED']),
-  ],
-  searchGeneralCourseName: [
-    ApiBearerAuth('accessToken'),
-    ApiOperation({
-      summary: '교양 과목명 강의 검색',
-      description: '교양 과목명을 입력하여 강의를 검색합니다.',
-    }),
-    ApiQuery({
-      name: 'courseName',
+      name: 'category',
       required: true,
-      type: 'string',
+      type: 'enum',
```

Comment:
```suggestion
      type: 'enum',
      enum: CourseCategory,
```
요거 넣어주면 swagger에서 요청이 잘 가는거 같습니다